### DISCUSSION TITLE
What to do when email address != IMAP/SMTP ID?

### DISCUSSION THREAD

**Discussion by giddyhup** [the-djmaze/snappymail#1092] [Q&A] [answered]
I plan to use SnappyMail within my Nextcloud setup. I cannot figure out what I need to configure in my case where the IMAP/SMTP ID is different from the email address.

If I am not mistaken, the login field expects an email address (or a user name which it would complete to an email addresse). Connecting with that to the IMAP server to fetch mail won't work in my case. I need some place to enter the connection user ID (a string without an '@') but I cannot get past the login.

Am I mssing something?

**Comment by cm-schl** [comment]
Hi,
I think the solution to your problem could be the flag "use short login" - see https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#imap-settings . Also this could help you: https://github.com/the-djmaze/snappymail/tree/master/integrations/nextcloud/snappymail#option-2-attempt-to-automatically-login-users-with-their-nextcloud-username-and-password-or-user-defined-credentials-if-set .
This describes the case where your Nextcloud username is the same as your imap username. If your config looks different it would be ideal to have more details about how you are logging in to Nextcloud and to your mail server 🙂

**Reply by giddyhup** [reply]
Thank you for your reply. I read through the provided sources and think they do not cover my situation. 

My Nextcloud user name is not the same as my email handle and the IMAP ID is different from both. In other words, while xy123456-2 is used together with a password for IMAP and SMTP, my email address is jdoe@mydomain.com and my Nextcloud user ID is UserY.

**Reply by cm-schl** [reply, marked as answer]
Ok in this situation I think you would have to:
- configure the autologin of Nextcloud to use the credentials manually set inside the settings of your Nextcloud user https://github.com/the-djmaze/snappymail/tree/master/integrations/nextcloud/snappymail#option-1-users-will-login-manually-or-define-credentials-in-their-personal-settings-for-automatic-logins. 
- Then, under additional settings of your Nextcloud user you could try to insert the credentials to use when logging in to your IMAP server.
- Inside of SnappyMail you have to define a default domain that will be added automatically at login https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#default-domain. This, in your case, will only be used to let SnappyMail know which domain configuration to use. 
- Inside the domain configuration you have to activate the flag to use short logins (to throw away the automatically added default domain) https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#add-domain.
Then you would have to create an identity containing your mail address to send out mails correctly.

I don't know if there are other (simpler) possibilities, but I think this could work...

**Reply by giddyhup** [reply]
I appreciate your patience and documentation links. I entered my (short) IMAP ID and my IMAP password in both login fields and got to access my mail. I had to make sure to enable the short address for both, SMTP and IMAP, and now it seems to be working also for sending. Thanks for guiding me!

**Reply by cm-schl** [reply]
You're welcome 🙂
